tests/admin_changelist/tests.py

```diff
@@ -860,6 +860,36 @@ def test_custom_lookup_with_pk_shortcut(self):
         cl = m.get_changelist_instance(request)
         self.assertCountEqual(cl.queryset, [abcd])
 
+    def test_search_with_exact_on_non_string_field(self):
+        """
+        ensures that the get_search_results method handles
+        the scenario where a non-integer term is provided
+        for an __exact search on an integer field.
+        """
+
+        child = Child.objects.create(name="Asher", age=11)
+        model_admin = ChildAdmin(Child, custom_site)
+
+        request = self.factory.get("/", data={SEARCH_VAR: "11"})
+        request.user = self.superuser
+        cl = model_admin.get_changelist_instance(request)
+        self.assertCountEqual(cl.queryset, [child])
+
+        request = self.factory.get("/", data={SEARCH_VAR: "Asher"})
+        request.user = self.superuser
+        cl = model_admin.get_changelist_instance(request)
+        self.assertCountEqual(cl.queryset, [child])
+
+        request = self.factory.get("/", data={SEARCH_VAR: "fsfsl"})
+        request.user = self.superuser
+        cl = model_admin.get_changelist_instance(request)
+        self.assertCountEqual(cl.queryset, [])
+
+        request = self.factory.get("/", data={SEARCH_VAR: "3.14"})
+        request.user = self.superuser
+        cl = model_admin.get_changelist_instance(request)
+        self.assertCountEqual(cl.queryset, [])
```

Comment:
```suggestion
    def test_search_with_exact_lookup_for_non_string_field(self):
        child = Child.objects.create(name="Asher", age=11)
        model_admin = ChildAdmin(Child, custom_site)

        for search_term, expected_result in [
            ("11", [child]),
            ("Asher", [child]),
            ("1", []),
            ("A", []),
            ("random", []),
        ]:
            request = self.factory.get("/", data={SEARCH_VAR: search_term})
            request.user = self.superuser
            with self.subTest(search_term=search_term):
                # 1 query for filtered result, 1 for filtered count, 1 for total count.
                with self.assertNumQueries(3):
                    cl = model_admin.get_changelist_instance(request)
                self.assertCountEqual(cl.queryset, expected_result)
```
I think using `subTest` and `assertNumQueries` would be nice here